self_supervised/utils.py

```diff
@@ -0,0 +1,27 @@
+# AUTOGENERATED! DO NOT EDIT! File to edit: nbs/00-utils.ipynb (unless otherwise specified).
+
+__all__ = ['transfer_weights']
+
+# Cell
+from fastai.vision.all import *
+
+# Cell
+def transfer_weights(learn:Learner, weights_path:Path, device:torch.device=None):
+    "Load and freeze pretrained weights inplace from `weights_path` using `device`"
+    if device is None: device = learn.dls.device
+    new_state_dict = torch.load(weights_path, map_location=device)
+    if 'model' in new_state_dict.keys(): new_state_dict = new_state_dict['model']
+    #allow for simply exporting the raw PyTorch model
+    learn_state_dict = learn.model.state_dict()
+    for name, param in learn_state_dict.items():
+        name = 'encoder.'+name[2:]
+        if name in new_state_dict:
+            input_param = new_state_dict[name]
+            if input_param.shape == param.shape:
+                param.copy_(input_param)
+            else:
+                print('Shape mismatch at:', name, 'skipping')
```

Comment:
I don't think skipping is the correct approach if there is mismatch, instead we should raise an exception.

Reply:
Anyway, I will go ahead and merge it. I can improve it later.

Reply:
Don't merge! I can do it easily real quick :)

Reply:
As I do agree, raising a `ValueError` would be better

Reply:
Cool! I am mainly thinking whether it would fail silently in any case. That would be the worse case! Thanks 👍

Reply:
Yeah let me play around with this a few minutes and work something out

Reply:
Alright this commit should fix that. We raise an exception if there is a weight mismatch as well as throw an error if no matching layers were found in general